### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by rod-glover**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Flask
+Flask-SQLAlchemy
+Flask-Cors
+PyCDS==2.0.1
+# Testing requirements
```
Comment (rod-glover) [inline requirements.txt]: Some of our repos include testing requirements in `requirements.txt`, others separate them into `test_requirements.txt`. This repo has both. We should decide which style we prefer.

File: sudo-user.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+adduser --disabled-password --gecos '' user
+adduser user sudo
+echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
```
Comment (rod-glover) [inline sudo-user.sh]: Should this be moved into `RUN` statements in `Dockerfile`?
Should this (in whatever form) be moved into the/a base image, e.g., `pcic/geospatial-python`?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+WORKDIR /app
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && \
+    apt-get -yq install postgresql postgresql-contrib postgis postgresql-9.3-postgis-2.1
+RUN apt list --installed
+RUN pip3 install -U pip
+RUN pip3 install -i https://pypi.pacificclimate.org/simple/ -r requirements.txt
+RUN python3 ./setup.py install
+RUN ./sudo-user.sh
```
Comment (rod-glover) [inline Dockerfile]: Should `sudo-user.sh` be inlined here? It's only 3 lines.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
+FROM pcic/geospatial-python
+
+ADD . /app
+WORKDIR /app
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && \
+    apt-get -yq install postgresql postgresql-contrib postgis postgresql-9.3-postgis-2.1
```
Comment (rod-glover) [inline Dockerfile]: Should the above ARG and RUN commands be moved into the `pcic/geospatial-python` package? They are foundational for any image that uses CRMP.

Comment (rod-glover) [reply Dockerfile]: Also, should we update `pcic/geospatial-python` to use Postgres 9.5, as our dev environments do? Then we'd have to install `postgresql-9.5-postgis-2.2`

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+language: python
+python:
+  - 2.7
+  - 3.4
+  - 3.5
+addons:
+  postgresql: "9.3"
+install:
+  - pip install -U pip
+  - pip install -i https://pypi.pacificclimate.org/simple/ -r requirements.txt -r test_requirements.txt
```
Comment (rod-glover) [inline .travis.yml:11]: We should decide whether we are going to have a separate `test_requirements.txt`. Currently test requirements are in both files.

File: wads/routes.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    :param map: the :class:`Map`.
+    """
+    regex = r'\d+'
+    num_convert = int
+
+    def __init__(self, map):
+        IntegerConverter.__init__(self, map, min=1850, max=2100)
+
+
```
Comment (rod-glover) [inline wads/routes.py]: Summarize what converters do

File: README.md [hunk 7]
```diff
@@ -1,2 +1,83 @@
...
+
+* Baseline data: Multi-year averages of:
+  * Monthly average of daily maximum temperature (tmax)
+  * Monthly average of daily minimum temperature (tmin)
+  * Monthly total precipitation (precip)
+* Weather data: For a given year and month:
+  * Monthly average of daily maximum temperature (tmax)
+  * Monthly average of daily minimum temperature (tmin)
+  * Monthly total precipitation (precip)
+
```
Comment (rod-glover) [inline README.md:21]: Explain response content

File: README.md [hunk 8]
```diff
@@ -1,2 +1,83 @@
...
+Invalid values for `<variable>`, `<year>`, or `<month>` result in a 404 Not Found.
+
+## Requirements
+
+libpq-dev(???) python-dev
+
+## Installation
+
+It is best practice to install using `virtualenv`.
```
Comment (rod-glover) [inline README.md]: Update to newer virtualenv built in

Comment (rod-glover) [reply README.md]: https://hangouts.google.com/hangouts/redirect?dest=https%3A%2F%2Fdocs.python.org%2F3%2Flibrary%2Fvenv.html

Comment (rod-glover) [reply README.md]: https://docs.python.org/3/library/venv.html

File: README.md [hunk 9]
```diff
@@ -1,2 +1,83 @@
...
+(venv)$ pip install -i http://pypi.pacificclimate.org/simple/ -e .
+```
+
+### Configuration
+
+Database dsn can be configured with the `CRMP_DSN` environment variable. 
+Defaults to `??? TBD`
+
+```bash
+(venv)$ CRMP_DSN=postgresql://dbuser:dbpass@dbhost/dbname scripts/devserver.py -p <port>
```
Comment (rod-glover) [inline README.md]: PCDS_DSN

File: requirements.txt (hunk 1 above)
Comment (rod-glover) [reply requirements.txt]: separate

File: setup.py [hunk 10]
```diff
@@ -0,0 +1,53 @@
+import sys
+from setuptools import setup, find_packages
+from setuptools.command.test import test as TestCommand
+
+__version__ = (0, 1, 0)
+
+
+class PyTest(TestCommand):
```
Comment (rod-glover) [inline setup.py]: Remove this class

File: wads/__init__.py [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+from flask.ext.cors import CORS
+
+from wads.routes import add_routes
+
+
+def get_app(config_override={}):
+    app = Flask(__name__)
+    CORS(app)
+    app.config['SQLALCHEMY_DATABASE_URI'] = os.getenv(
+        'CRMP_DSN', 'postgresql://httpd_meta@monsoon.pcic.uvic.ca/crmp') # FIXME: Correct default database URI
```
Comment (rod-glover) [inline wads/__init__.py]: PCDS_DSN

Comment (rod-glover) [reply wads/__init__.py]: username: httpd

File: wads/__init__.py [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import os
+
+from flask import Flask
+from flask.ext.cors import CORS
```
Comment (rod-glover) [inline wads/__init__.py]: import from flask_cors

Comment (rod-glover) [reply wads/__init__.py]: remove CORS

File: wads/api/baseline.py [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+    """
+
+    db_variable_name = {
+        'tmax': 'Tx_Climatology',
+        'tmin': 'Tn_Climatology',
+        'precip': 'Precip_Climatology',
+    }
+
+    q = session.query(
+        History.station_name,
```
Comment (rod-glover) [inline wads/api/baseline.py]: add network name, station native id

File: wads/api/weather.py [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+    view_for_variable = {
+        'tmax': MonthlyAverageOfDailyMaxTemperature,
+        'tmin': MonthlyAverageOfDailyMinTemperature,
+        'precip': MonthlyTotalPrecipitation,
+    }
+
+    WeatherView = view_for_variable[variable]
+
+    q = session.query(
+        History.station_name,
```
Comment (rod-glover) [inline wads/api/weather.py]: add network name, station native id

File: wads/api/weather.py [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+        History.lon,
+        History.lat,
+        History.elevation,
+        # TODO: Do we need other History info, e.g., frequency?
+        # TODO: Do we need other Variable info?
+        Variable.cell_method,
```
Comment (rod-glover) [inline wads/api/weather.py]: History.freq
Variable.network_var_name

File: tests/test_routes.py [hunk 16]
```diff
@@ -0,0 +1,50 @@
+from pytest import mark
+import json
+import wads.api
+
+
```
Comment (rod-glover) [inline tests/test_routes.py:13]: Explain mocking

File: .travis.yml (hunk 5 above)
Comment (rod-glover) [reply .travis.yml:11]: Done: separate files.

**Comment by rod-glover** [comment]
Tests continue to fail because PyCDS is not up to date. Tests pass on my dev environment with latest PyCDS branch.

**Review by jameshiebert — APPROVED** [review]
We've been running this for a couple of months now, so we may as well merge it!

**Pull request merged**